joomla/joomla-cms
# [4.0] Adding testing sampledata plugin

This adds the testing sampledata plugin. Contentwise this is a copy of https://github.com/joomla-extensions/testing-sample-data/, but I refactored the code to work in 4.0. This also is not ready yet. The support for workflow needs to be added as far as I can tell and the content in generell needs to be overhauled. It has become a bit dated...

### How to test
Checkout this branch, install Joomla like regular, discover the testing plugin and install it, then enable it. Now you can see the testing sampledata entry in the home dashboard. Click on install and wait for it to do its thing. 

### How can this be included in Joomla?
I think this plugin should be part of the Joomla main repo to ease testing, but of course it shouldn't be part of the actual release. So it should be deleted from the release package during building the release and the SQL changes necessary to have this automatically available during development time. Deleting the files is pretty easy, the SQL part is what is making this difficult. I'm considering changing the installation to run the database setup step in several sub-steps, one of them could be the testing plugin. Every step could be a SQL file and for the release the SQL for the testing plugin is deleted, which would automatically drop that step. Lets see if I can get that part done...
